src/Api/ApiServiceProvider.php

```diff
@@ -49,6 +49,8 @@ public function register()
         $this->app->singleton('flarum.api.middleware', function (Application $app) {
             $pipe = new MiddlewarePipe;
 
+            $pipe->pipe($app->makeWith(HttpMiddleware\MaintenanceMode::class, ['maintenance' => $app->isDownForMaintenance()]));
```

Comment:
I didn't know `makeWith()` [existed](https://github.com/laravel/framework/blob/5.7/src/Illuminate/Container/Container.php#L588-L598), interesting. :grinning: 

Since it's not part of the contract, could you just use `make()` instead?

Reply:
Odd 🤔 I must have looked it up in the docs which mentions makeWith instead of make:

![image](https://user-images.githubusercontent.com/504687/53961160-407c7780-40e8-11e9-9505-5856c9fd7820.png)

Reply:
I think maybe there was some point at which Taylor wanted to remove the ability to pass params through the `make` API, and added a `makeWith` method to keep people who rely on that functionality happy. But there was backlash so it got reverted?